### ISSUE TITLE
Return more information in smarts.reset

### ISSUE THREAD

**Issue by JingfeiPeng** [huawei-noah/SMARTS#421]
Currently smarts.reset only returns `observations_for_ego`. It would be helpful for smarts.reset to return all 4 pieces of information `observations, rewards, dones, extras` that it got in smarts.step(). This is an interface change and env files and some test files need to change how they handle the return value from smarts.reset.

Smarts.py Line 321
```
        if len(self._agent_manager.ego_agent_ids):
            while len(observations_for_ego) < 1:
                observations_for_ego, _, _, _ = self.step({})

        self._reset_providers()

        return observations_for_ego
```

Related issue #54

**Comment by Gamenot** [comment]
Based off of issue #54

**Comment by Gamenot** [comment]
Instead of this we should skip to the next episode and give a warning to conform to gym API.

**Comment by Gamenot** [comment]
Was closed by #397.
